doc/troubleshoot-checks.md

```diff
@@ -0,0 +1,135 @@
+
+# Built-in troubleshooting functionality
+The built-in troubleshooting functionality in the `iotedge` CLI, "iotedge check", performs configuration and connectivity checks for commonly encountered issues.
+
+`iotedge help check` displays detailed usage information.
+
+# Scope
+
+The troubleshooting tool is focused on
+
+* Surfacing potential problems that prevent getting Edge device cloud connected.
+
+* Surfacing potential configuration deviations from recommended production best-practices.
+
+By design, it does not check for errors in the Edge workload deployment. Examples for these are access to any private container registry used, errors in module create options etc. The rationale being deployment validation is best performed in the facility where it is authored.
+
+Checks that would involve parsing IoT Edge module logs or metrics are also out of scope.
+
+# Result types
+
+Results from checks are characterized as either **errors** or **warnings**.
+
+Errors have a high likelihood of preventing the Edge runtime or the modules from connecting to the cloud.
+
+Warnings might not affect immediate connectivity but are potential deviations from best practices and may affect long term stability, offline operation or supportability of the Edge device.
+
+# Configuration checks details
+
+## config.yaml is well-formed
+
+IoT Edge's config.yaml is valid and free of any syntax (e.g. whitespace) errors. Depending on the error, it may not exactly be at the line number and position reported by the parser.
```

Comment:
```suggestion
IoT Edge's `config.yaml` is valid and free of any syntax (e.g. whitespace) errors. Depending on the error, it may not exactly be at the line number and position reported by the parser.
```